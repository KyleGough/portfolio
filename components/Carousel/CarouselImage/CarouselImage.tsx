import { clsx } from 'clsx';
import React from 'react';

interface CarouselImageProps {
  show: boolean;
  bg: string;
}

export const CarouselImage: React.FC<CarouselImageProps> = ({ show, bg }) => (
  <div
    className={clsx(
      bg,
      { 'opacity-100': show },
      { 'opacity-0': !show },
      'absolute top-0 left-0',
      'w-full h-full',
      'bg-fixed bg-center bg-cover',
      'duration-2000 transition-opacity'
    )}
  ></div>
);
