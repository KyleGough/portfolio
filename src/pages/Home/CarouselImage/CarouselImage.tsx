import React from 'react';

interface CarouselImageProps {
  show: boolean;
  className?: string;
}

export const CarouselImage: React.FC<CarouselImageProps> = ({
  show,
  className,
}) => (
  <div
    className={`${className} ${
      show ? 'opacity-100' : 'opacity-0'
    } absolute top-0 left-0 w-full h-full bg-fixed bg-center bg-cover duration-2000 transition-opacity`}
  ></div>
);
