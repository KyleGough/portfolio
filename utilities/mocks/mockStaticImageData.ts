const mockStaticImageData = {
  src: '/#image',
  width: 100,
  height: 100,
  blurDataURL: '/#blur',
};

export default mockStaticImageData;
