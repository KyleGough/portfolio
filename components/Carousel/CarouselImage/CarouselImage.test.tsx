import mockStaticImageData from '@mocks/mockStaticImageData';
import { render } from '@testing-library/react';
import React from 'react';

import { CarouselImage } from './CarouselImage';

describe('CarouselImage component', () => {
  it('renders', () => {
    const { container } = render(
      <CarouselImage show={true} bg={mockStaticImageData} />
    );
    expect(container).toMatchSnapshot();
  });
});
