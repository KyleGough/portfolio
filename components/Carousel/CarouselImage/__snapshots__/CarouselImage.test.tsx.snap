// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselImage component renders 1`] = `
<div>
  <div
    class="bg-header1 opacity-100 absolute top-0 left-0 w-full h-full bg-fixed bg-center bg-cover duration-2000 transition-opacity"
  />
</div>
`;
